include/exadg/time_integration/time_int_bdf_base.h

```diff
@@ -101,6 +101,15 @@ class TimeIntBDFBase : public TimeIntBase
   double
   get_previous_time(int const i /* t_{n-i} */) const;
 
+  /*
+   * Get current order of time integrator
+   */
+  unsigned int
+  get_current_order() const;
+
+  BDFTimeIntegratorConstants const &
+  get_bdf_constants() const;
```

Comment:
From the current PR, I do not see that these should be public.

Reply:
I agree on `get_current_order()`. This should be private. As mention in the description, I want to use this infrastructure to compute the temporal derivative of the pressure outside the `INSE` module. So I need `get_bdf_constants()` to stay public :)

Reply:
Hm, I feel this should not be public information. A main problem that I see is that outside the class one can not know whether the BDF constants are correct, because they are updated in case of adaptive time step sizes. Things are more encapsulated if this happens within the time integrator.

Reply:
Or is it OK to have `compute_dp_dt()` in the BDF time integrator of the INSE?

Reply:
Yes, this is what I would like to discuss: In terms of encapsulation, is it better (i) that the time integrator computes a temporal derivative when requested, or (ii) that the time integrator returns time integration constants (which is somewhat "private" information)?

I tend to prefer the first option. What do you think?

Reply:
I also prefer the first one.
Just one note: To always get the temporal derivatives correct, we have to do some more work. This is because pressure_np might already be pushed back or not. Therefore, I would like to keep track of the times in the same way we keep track of the solution vectors and check the order before computing the derivative. Is this OK for you?

Reply:
Yes, I think such a check is the advantage of the first option over the second one. Right now, it is not fully clear to me what you mean by "check the order".

I think one should pass the time t at which to evaluate the temporal derivative to that public member function of the time integrator, and then check inside that function whether this time t is equivalent to time t_np stored by the time integrator (up to some tolerance).
The assertion that time instants and solution vectors of the time integrator are in sync is an independent responsibility of the time integrator (not directly related to the functionality discussed here).

Reply:
So I am principally OK with this. If this requires fundamental changes of the time integrators, we should discuss beforehand whether to address these basic changes in a separate PR.

Reply:
> Yes, I think such a check is the advantage of the first option over the second one. Right now, it is not fully clear to me what you mean by "check the order".

With order, I mean sequence of solutions in the vector.

> I think one should pass the time t at which to evaluate the temporal derivative to that public member function of the time integrator, and then check inside that function whether this time t is equivalent to time t_np stored by the time integrator (up 
to some tolerance). 

Fully agree.

The assertion that time instants and solution vectors of the time integrator are in sync is an independent responsibility of the time integrator (not directly related to the functionality discussed here).

I would still like to do it in the PR if it is OK. The whole point, I need introduced code, is just to compute temporal pressure derivatives. Is it OK for you if this is also done in this PR?

Reply:
> So I am principally OK with this. If this requires fundamental changes of the time integrators, we should discuss beforehand whether to address these basic changes in a separate PR.

I think this only requires minor changes in the `INSE::BDFTimeIntegrator`.

Reply:
@nfehn To achieve this, I have to keep track of the time the same way we do it for the vector of solutions. Since `solution_np` is set in `do_timestep_solve()` also `time_history_np` should be set in this function. I added a draft in [385a46b](https://github.com/exadg/exadg/pull/458/commits/385a46b2d47f6f81a9ca3077fbeb4546c736b654). Can you relate to these changes? If yes, I will rebase and split the functions to header and source files.